### PR TITLE
feature: TEMPERATURE_WAIT instead of m109

### PR DESCRIPTION
**Pull request by krautech**
## Description

Solves Issue #64 

## Checklist

The following relevant macros have been tested:

- [ ] `Has been tested while printing?`

### CODE REVIEW

File: scanner.py [hunk 1]
```diff
@@ -604,7 +605,7 @@ def set_temp(self,gcmd):
             cur_temp = hotend.get_heater().get_status(curtime)["temperature"]
             if self.extruder_target > cur_temp:
                 gcmd.respond_info("Heating hotend to %.1f" % (self.extruder_target))
-                cmd = "M109 S"+str(self.extruder_target)
+                cmd = f"TEMPERATURE_WAIT SENSOR=extruder MAXIMUM="+str(self.extruder_target)
```
[the remark this answers is not in the document]
Comment (krautech) [reply scanner.py:608]: The way I have it tells it to resume heating back to the initial temperature that was set and wait until it is before proceeding with the rest of your print start macro

[the remark this answers is not in the document]
Comment (krautech) [reply scanner.py:608]: Actually you're right 👍🏼 derp.

**Review by Jomik — COMMENTED** [review]
I am unsure if it is possible to name it something other than `extruder`?
I noticed that `hotend.get_heater()` should have a `get_name()` function?
https://github.com/Klipper3d/klipper/blob/9bd0d4757696375e946a1410fcc8036d2670ae30/klippy/extras/heaters.py#L96
So maybe we should use that?

It seems like you could have `extruder0`, `extruder1` and so on 
🤔  Don't know if it actually matters.

**Comment by krautech** [comment]
> I am unsure if it is possible to name it something other than `extruder`? I noticed that `hotend.get_heater()` should have a `get_name()` function? https://github.com/Klipper3d/klipper/blob/9bd0d4757696375e946a1410fcc8036d2670ae30/klippy/extras/heaters.py#L96 So maybe we should use that?
> 
> It seems like you could have `extruder0`, `extruder1` and so on 🤔 Don't know if it actually matters.

I dont think its worth going in that deep. Cartographer only works with one toolhead per printer in its current state anyway.

**Review by Jomik — APPROVED** [review]

**Pull request merged**